### PR TITLE
Km test correlation

### PR DESCRIPTION
**Pull request by kenmawer**
This is to test the `epi_cor` and `shift` functions.

### CODE REVIEW

File: R/archive.R [hunk 1]
```diff
@@ -178,10 +178,15 @@ epi_archive =
             cat(sprintf("* %-14s = %s\n", "max version",
                         max(self$DT$version)))
             cat("----------\n")
-            cat(sprintf("Data archive (stored in DT field): %i x %i\n",
+            cat(sprintf("Data archive (stored in DT field): %i x %i\n", 
                         nrow(self$DT), ncol(self$DT)))
             cat("----------\n")
-            cat(sprintf("Public methods: %s",
+            cat(sprintf("Columns in DT: %s\n", paste(ifelse(length(
```
Comment (ChloeYou) [inline R/archive.R:184]: seems like this is the part that was changed compared to main.

File: R/slide.R [hunk 2]
```diff
@@ -84,7 +84,27 @@
 #'   tidy evaluation (first example, above), then the name for the new column is 
 #'   inferred from the given expression and overrides any name passed explicitly 
 #'   through the `new_col_name` argument.
-#' 
+#'
```
Comment (ChloeYou) [inline R/slide.R:87]: added more description.

File: R/slide.R [hunk 3]
```diff
@@ -168,6 +188,10 @@ epi_slide = function(x, f, ..., n = 7, ref_time_values,
   time_range = range(unique(x$time_value))
   starts = in_range(ref_time_values - before_num, time_range)
   stops = in_range(ref_time_values + after_num, time_range)
+
```
Comment (ChloeYou) [inline R/slide.R:191]: added new description.

**Review by ChloeYou — CHANGES_REQUESTED** [review]

File: tests/testthat/test-correlation.R [hunk 4]
```diff
@@ -0,0 +1,23 @@
+library(tibble)
+
+test_that("epi_cor throws errors as needed",{
```
[the remark this answers is not in the document]
Comment (kenmawer) [reply tests/testthat/test-correlation.R]: I changed this to be more explanatory.

File: tests/testthat/test-correlation.R [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  expect_equal(epi_cor(x = jhu_csse_daily_subset, 
+               var1 = case_rate_7d_av, 
+               var2 = death_rate_7d_av, 
+               cor_by = geo_value, 
+               dt1 = -2)[1],
+               tibble(geo_value = unique(jhu_csse_daily_subset$geo_value))
+  )
+})
+
+test_that("shift works as it should",{
```
[the remark this answers is not in the document]
Comment (kenmawer) [reply tests/testthat/test-correlation.R]: I fixed these.

**Review by ChloeYou — APPROVED** [review]

**Pull request merged**